src/k2/dto/FieldTypes.cpp

```diff
@@ -33,51 +33,97 @@ Copyright(c) 2020 Futurewei Cloud
 namespace k2 {
 namespace dto {
 
-static constexpr char TERM = '\0';
+static constexpr char ESCAPE = '\0';
+static constexpr char TERM = 0x01;
+static constexpr char ESCAPED_NULL = 0xFF;
 
 template <> FieldType TToFieldType<String>() { return FieldType::STRING; }
 template <> FieldType TToFieldType<uint32_t>() { return FieldType::UINT32T; }
 
 // All conversion assume ascending ordering
 
-template <> String FieldToKeyString<String>(const String& field) {
-    // Sizes will not match if there are exta null bytes
-    K2ASSERT(field.size() == strlen(field.c_str()), "String has null bytes");
+static String escapeKeyString(const String& field) {
+    size_t pos = 0;
+    std::vector<size_t> foundNulls;
+
+    while (pos != String::npos && pos < field.size()) {
+        pos = field.find(ESCAPE, pos);
+        if (pos != String::npos) {
+            foundNulls.push_back(pos);
+            ++pos;
+        }
+    }
+
```

Comment:
maybe shortcut the function here: if no nulls were found just return the original